### PR TITLE
Add tags on endpoints

### PR DESCRIPTION
**Pull request by pm3310**
*Issue #, if available:*

*Description of changes:*

Option to add Tags on SageMaker Endpoints. I'd like this option so that to tag endpoints for auditing and linking IDs of trained models to endpoints. The latter is very important because I can know which models are dpeloyed in production.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated the [changelog](https://github.com/aws/sagemaker-python-sdk/blob/master/CHANGELOG.rst) with a description of my changes (if appropriate)
- [ ] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by laurenyu** [comment]
the integ tests are failing with this error:

```
        with timeout_and_delete_endpoint_by_name(endpoint_name, sagemaker_session):
            estimator = Estimator.attach(training_job_name=training_job_name, sagemaker_session=sagemaker_session)
            model = estimator.create_model()
>           predictor = model.deploy(1, 'ml.m4.xlarge', endpoint_name=endpoint_name)

tests/integ/test_byo_estimator.py:154: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.tox/py27/lib/python2.7/site-packages/sagemaker/model.py:102: in deploy
    self.sagemaker_session.endpoint_from_production_variants(self.endpoint_name, [production_variant], tags)
.tox/py27/lib/python2.7/site-packages/sagemaker/session.py:665: in endpoint_from_production_variants
    EndpointConfigName=name, ProductionVariants=production_variants, Tags=tags)
.tox/py27/lib/python2.7/site-packages/botocore/client.py:314: in _api_call
    return self._make_api_call(operation_name, kwargs)
.tox/py27/lib/python2.7/site-packages/botocore/client.py:586: in _make_api_call
    api_params, operation_model, context=request_context)
.tox/py27/lib/python2.7/site-packages/botocore/client.py:621: in _convert_to_request_dict
    api_params, operation_model)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <botocore.validate.ParamValidationDecorator object at 0x7fa197e2bb50>
parameters = {'EndpointConfigName': 'byo-2018-06-25-16-42-17-111', 'ProductionVariants': [{'InitialInstanceCount': 1, 'InitialVaria... 1, 'InstanceType': 'ml.m4.xlarge', 'ModelName': 'factorization-machines-2018-06-25-16-46-03-730', ...}], 'Tags': None}
operation_model = OperationModel(name=CreateEndpointConfig)

    def serialize_to_request(self, parameters, operation_model):
        input_shape = operation_model.input_shape
        if input_shape is not None:
            report = self._param_validator.validate(parameters,
                                                    operation_model.input_shape)
            if report.has_errors():
>               raise ParamValidationError(report=report.generate_report())
E               ParamValidationError: Parameter validation failed:
E               Invalid type for parameter Tags, value: None, type: <type 'NoneType'>, valid types: <type 'list'>, <type 'tuple'>
```

Can you modify your change to not pass along `tags` to the SageMaker call if it's `None`?

**Comment by pm3310** [comment]
Hey @laurenyu , I fixed it :-)

File: src/sagemaker/session.py [hunk 1]
```diff
@@ -660,8 +661,12 @@ def endpoint_from_production_variants(self, name, production_variants, wait=True
...
         if not _deployment_entity_exists(
                 lambda: self.sagemaker_client.describe_endpoint_config(EndpointConfigName=name)):
-            self.sagemaker_client.create_endpoint_config(
-                EndpointConfigName=name, ProductionVariants=production_variants)
+            if tags:
+                self.sagemaker_client.create_endpoint_config(
+                    EndpointConfigName=name, ProductionVariants=production_variants, Tags=tags)
+            else:
+                self.sagemaker_client.create_endpoint_config(
+                    EndpointConfigName=name, ProductionVariants=production_variants)
```
Comment (laurenyu) [inline src/sagemaker/session.py]: what do you think of doing it this way?

```python
config_options = {'EndpointConfigName': name, 'ProductionVariants': production_variants}

if tags:
    config_options['Tags'] = tags

self.sagemaker_client.create_endpoint_config(**config_options)
```

**Comment by pm3310** [comment]
@laurenyu great idea. Done!

**Pull request merged**